app/src/main/java/org/wikipedia/search/SearchResultsFragment.java

```diff
@@ -322,13 +335,32 @@ private void doFullTextSearch(final String searchTerm,
 
                     // full text special:
                     SearchResultsFragment.this.lastFullTextResults = results;
-                    displayResults(resultList);
+                    if (!resultList.isEmpty()) {
+                        displayResults(resultList);
+                    }
+                    return resultList.isEmpty() ? doFullTextSearchResultsCountObservable(searchTerm) : Observable.empty();
+                })
+                .toList()
+                .doAfterTerminate(() -> updateProgressBar(false))
```

Comment:
When you return an `empty()` observable, it looks like this part is never called, and the progress bar keeps spinning forever.

Reply:
I put `L.d()` in `doAfterTerminate()` and `subscribe()`, both of the logs will show on the Logcat. I believe the `toList()` will create an empty list for the `empty()` and it will not block the flow. 

If I remove the `toList()`, the flow will not go to `subscribe()` but will actually jump to the `doAfterTerminate()`.

BTW, there's a `Observable.never()` will block the flow that includes the `doAfterTerminate()`.